lib/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -373,9 +374,53 @@ public static function get_theme_data( $deprecated = array(), $options = array()
 		$with_theme_supports = new WP_Theme_JSON_Gutenberg( $theme_support_data );
 
 		if ( $options['with_block_style_variations'] ) {
+			// Absorb block style variations that were registered with a style object.
 			$block_style_variations_data = WP_Theme_JSON_Gutenberg::get_from_block_styles_registry();
 			$with_block_style_variations = new WP_Theme_JSON_Gutenberg( $block_style_variations_data );
 			$with_theme_supports->merge( $with_block_style_variations );
+
+			// Resolve shared block style variations that were bundled in the
+			// theme via standalone theme.json files.
+			$shared_block_style_variations = static::get_style_variations( '/block-styles' );
+			$variations_data               = array();
+			$registry                      = WP_Block_Styles_Registry::get_instance();
+
+			foreach ( $shared_block_style_variations as $variation ) {
+				if ( empty( $variation['supportedBlockTypes'] ) || empty( $variation['styles'] ) ) {
+					continue;
+				}
+
+				$variation_slug = _wp_to_kebab_case( $variation['title'] );
+
+				// If it proves desirable, block style variations could include
+				// custom settings which can be included here.
+				foreach ( $variation['supportedBlockTypes'] as $block_type ) {
+					// Automatically register the block style variation if it
+					// hasn't been already.
+					$registered_styles = $registry->get_registered_styles_for_block( $block_type );
+					if ( ! array_key_exists( $variation_slug, $registered_styles ) ) {
+						gutenberg_register_block_style(
+							$block_type,
+							array(
+								'name'  => $variation_slug,
+								'label' => $variation['title'],
+							)
+						);
+					}
```

Comment:
Should block style variations defined within these standalone files be registered for blocks automatically? 

Child themes wouldn't be able to deregister block style variations coming from standalone files via `unregister_block_style` but they could override the variation setting empty `styles` or `supportedBlockTypes` properties to avoid the automatic registration and omit that variation.

Reply:
1) I would prefer that, yes.

2) This is currently how child themes need to override (remove) theme style variations, so it makes sense.

Yes it is a little limiting, but it is already documented that "The function unregister_block_style only unregisters styles that were registered on the server using register_block_style" so there would be no change.